### ISSUE TITLE
Suggestions for usage scenarios

### ISSUE THREAD

**Issue by aaronchn** [ahsifer/goxdp#2]
For me, the expected usage scenario for goxdp is this:

I would like to first create an allow address rule, it would be nice if I could also specify the port, in which case the first rule would be allow 1.1.1.1:22, which would ensure that I don't lock myself out of the house when debugging the rule.

Next create a second rule about address. block 0.0.0.0/0, which will ensure that unless explicitly allowed (like the first one already), all visits will be denied.

Next create a third or more rules that are port oriented, such as 80. this will ensure that an address, after being allowed access, continues to determine whether access to a specific port is allowed.
In summary, deny a large number of addresses and allow a small number of addresses. With access allowed, it continues to determine if it is a permitted port or not, otherwise it is denied.

Obviously, goxdp has shown potential in this area, and look forward to seeing features that apply to the above scenarios.

Also, there are already projects or packages that use the goxdp name, so perhaps an individual name would be a better choice.

Thanks
